### PR TITLE
ceph: Retry mirror status update in tests

### LINKED ISSUE #6736: ci: TestCephBlockPoolController unit test randomly fails
Here is the test:

```
2020-12-02T14:25:16.5472707Z --- FAIL: TestCephBlockPoolController (0.01s)
2020-12-02T14:25:16.5473400Z     controller_test.go:357: 
2020-12-02T14:25:16.5473945Z         	Error Trace:	controller_test.go:357
2020-12-02T14:25:16.5474505Z         	Error:      	Should NOT be empty, but was 
2020-12-02T14:25:16.5475213Z         	Test:       	TestCephBlockPoolController
2020-12-02T14:25:16.5475865Z         	Messages:   	map[]
2020-12-02T14:25:16.5476319Z     controller_test.go:361: 
2020-12-02T14:25:16.5476856Z         	Error Trace:	controller_test.go:361
2020-12-02T14:25:16.5477448Z         	Error:      	Received unexpected error:
2020-12-02T14:25:16.5478118Z         	            	secrets "" not found
2020-12-02T14:25:16.5478782Z         	Test:       	TestCephBlockPoolController
2020-12-02T14:25:16.5479733Z panic: runtime error: invalid memory address or nil pointer dereference [recovered]
2020-12-02T14:25:16.5480594Z 	panic: runtime error: invalid memory address or nil pointer dereference
2020-12-02T14:25:16.5481362Z [signal SIGSEGV: segmentation violation code=0x1 addr=0x120 pc=0x1472927]
2020-12-02T14:25:16.5481792Z 
2020-12-02T14:25:16.5482155Z goroutine 22 [running]:
2020-12-02T14:25:16.5482674Z testing.tRunner.func1(0xc0004fca00)
2020-12-02T14:25:16.5483506Z 	/opt/hostedtoolcache/go/1.13.15/x64/src/testing/testing.go:874 +0x3a3
2020-12-02T14:25:16.5484073Z panic(0x15da880, 0x25ec3d0)
2020-12-02T14:25:16.5484679Z 	/opt/hostedtoolcache/go/1.13.15/x64/src/runtime/panic.go:679 +0x1b2
2020-12-02T14:25:16.5485643Z github.com/rook/rook/pkg/operator/ceph/pool.TestCephBlockPoolController(0xc0004fca00)
2020-12-02T14:25:16.5486752Z 	/home/runner/go/src/github.com/rook/rook/pkg/operator/ceph/pool/controller_test.go:362 +0x1a27
2020-12-02T14:25:16.5487487Z testing.tRunner(0xc0004fca00, 0x186a2f8)
2020-12-02T14:25:16.5488166Z 	/opt/hostedtoolcache/go/1.13.15/x64/src/testing/testing.go:909 +0xc9
2020-12-02T14:25:16.5488754Z created by testing.(*T).Run
2020-12-02T14:25:16.5489381Z 	/opt/hostedtoolcache/go/1.13.15/x64/src/testing/testing.go:960 +0x350
2020-12-02T14:25:16.5490057Z FAIL	github.com/rook/rook/pkg/operator/ceph/pool	0.031s
```

### PR DESCRIPTION
**Pull request by travisn**
**Description of your changes:**
The TestCephBlockPoolController test is failing intermittently due to a failure to update the rbd mirroring status on the pool CR. Improve the test reliability by trying again several times if needed.

**Which issue is resolved by this Pull Request:**
Resolves #6736 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: pkg/operator/ceph/pool/controller_test.go [hunk 1]
```diff
@@ -348,19 +348,27 @@ func TestCephBlockPoolController(t *testing.T) {
...
+		res, err = r.Reconcile(req)
+		assert.NoError(t, err)
+		assert.False(t, res.Requeue)
+		err = r.client.Get(context.TODO(), req.NamespacedName, pool)
+		assert.NoError(t, err)
+		assert.Equal(t, cephv1.ConditionReady, pool.Status.Phase)
+		if _, ok := pool.Status.Info[RBDMirrorBootstrapPeerSecretName]; ok {
+			break
+		}
+		logger.Infof("FIX: trying again to update the mirroring status")
```
Comment (travisn) [inline pkg/operator/ceph/pool/controller_test.go:361]: There is still an issue that needs investigating with the mirroring status not being updated on the pool CR, the intention of this PR is to at least make the test more stable until we find the root cause.

Comment (subhamkrai) [reply pkg/operator/ceph/pool/controller_test.go:361]: Looks good to make CI pass. I'm looking into this and trying to find the root cause.

**Review by subhamkrai — APPROVED** [review]

**Comment by BlaineEXE** [comment]
I'd feel better waiting for a root cause fix, but this does look good as code.

**Comment by travisn** [comment]
> I'd feel better waiting for a root cause fix, but this does look good as code.

The intermittent unit test failure really blocks the CI too often. As an example, on Friday it took the 4th run of the CI to get the release build out, which really slowed down the urgent release. I'd rather merge and leave the issue opened to investigate independently of blocking the build.

**Review by BlaineEXE — APPROVED** [review]

**Pull request merged**